cmake/MOAB_PullAndMake.cmake

```
@@ -0,0 +1,86 @@
+# This Macro sets up the download and build of MOAB using ExternalProject
+# few tweak are done in src/dagmc/CMakeLists.txt and src/PyNE/CMakeists.txt 
+# to make sure that MOAB is built before DAGMC.
+MACRO (moab_pull_make MOAB_Version)
+  MESSAGE(STATUS "MOAB will be downloaded and built")
+  
+  # Ensure STATIC LIBS Build
+  IF(DAGMC_BUILD_STATIC_LIBS)
+    MESSAGE(FATAL_ERROR "PULL_INSTALL_MOAB is ONLY compatible with shared libraries.")
+  endIF()
+    
+  # Ensure we have the right variables to build MOAB
+  moab_autobuild_check_deps()
+   
+    
+  INCLUDE(ExternalProject)
+  MESSAGE("HDF5_ROOT: ${HDF5_ROOT}")
+  SET(MOAB_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/moab")
+  SET(MOAB_ROOT "${CMAKE_BINARY_DIR}/moab")
+  SET(MOAB_INCLUDE_DIRS "${MOAB_INSTALL_PREFIX}/include")
+  SET(MOAB_LIBRARY_DIRS "${MOAB_INSTALL_PREFIX}/lib")
+  MESSAGE("MOAB_LIBRARY_DIRS: ${MOAB_LIBRARY_DIRS}")
+  MESSAGE("CMAKE_SHARED_LIBRARY_SUFFIX: ${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  SET(MOAB_LIBRARIES_SHARED "")
+  ExternalProject_Add(MOAB_ep
+    PREFIX ${MOAB_ROOT}   
+    GIT_REPOSITORY https://bitbucket.org/fathomteam/moab.git
+    GIT_TAG ${moab_version}
+    CMAKE_ARGS
+      -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+      -DCMAKE_BUILD_TYPE=Release
```

I think it makes sense for us to pass the same value for the build type that was provided for DAGMC. If one is looking to debug a problem with DAGMC, they'd likely want MOAB to be build in debug mode as well.

```suggestion
      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
```

This raises perhaps another question though: What will happen if DAGMC is recompiled with a new value of `CMAKE_BUILD_TYPE` will the MOAB build reflect that change?

---

This was marked as resolved but I don't see an update to the line above. Am I missing something?

---

PR was out of date (though no `Refresh` was apparent on the page). Mea culpa. Disregard.